casper/src/main/scala/coop/rchain/casper/dag/BlockDagKeyValueStorage.scala

```diff
@@ -57,6 +58,35 @@ final class BlockDagKeyValueStorage[F[_]: Concurrent: Log] private (
         deployHashes = block.state.deploys.map(_.deploy.sig)
         _            <- deployIndex.put(deployHashes.map(_ -> block.blockHash))
 
+        dagState <- representationState.get.map(_.dagMessageState)
+
+        // Store fringe data
+        fringeHash = FringeData.fringeHash(blockMetadata.fringe)
+        // Calculate blocks included in the fringe
+        finalizer          = Finalizer(dagState.msgMap)
+        justificationsMsgs = blockMetadata.justifications.map(dagState.msgMap)
+        prevFringeMsgs     = finalizer.latestFringe(justificationsMsgs)
+        fringeMsgs         = blockMetadata.fringe.map(dagState.msgMap)
+        fringeDiff         = fringeMsgs.flatMap(_.seen) -- prevFringeMsgs.flatMap(_.seen)
```

Comment:
Should this be include `fringeMsgs  ++ fringeMsgs.flatMap(_.seen) -- prevFringeMsgs -- prevFringeMsgs.flatMap(_.seen)` ?